controllers/group_totals.go

```diff
@@ -56,48 +54,132 @@ func InstaCountHandler(rw http.ResponseWriter, r *http.Request) {
 		http.Error(rw, err.Error(), http.StatusInternalServerError)
 	}
 
-	queryType := mux.Vars(r)["type"]
-	var pipeline models.Pipeline
-	switch queryType {
-	case "patient":
-		pipeline = models.NewPipeline(group)
-	case "encounter":
-		pipeline = models.NewEncounterPipeline(group)
-	case "condition":
-		pipeline = models.NewConditionPipeline(group)
-	}
-	qr, err := pipeline.ExecuteCount(server.Database)
-	if err != nil && err != mgo.ErrNotFound {
-		http.Error(rw, err.Error(), http.StatusInternalServerError)
-		return
+	pquery := ""
+	cquery := ""
+	equery := ""
+
+	intersectRequired := false
+
+	for _, characteristic := range group.Characteristic {
+
+		codings := characteristic.Code.Coding
+
+		for _, coding := range codings {
+			//Age
+			if coding.System == "http://loinc.org" && coding.Code == "21612-7" {
+				highAgeDate := time.Date(time.Now().Year()-int(*characteristic.ValueRange.Low.Value), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC).Format("2006-01-02")
+				lowAgeDate := time.Date(time.Now().Year()-int(*characteristic.ValueRange.High.Value), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC).Format("2006-01-02")
+				pquery += "birthdate=lt" + highAgeDate + "&birthdate=gt" + lowAgeDate + "&"
+				cquery += "patient.birthdate=lt" + highAgeDate + "&patient.birthdate=gt" + lowAgeDate + "&"
+				equery += "patient.birthdate=lt" + highAgeDate + "&patient.birthdate=gt" + lowAgeDate + "&"
```

Comment:
I'm not sure this logic is quite right.  For example, if the age range is 18 - 65, and today is the patient's 18th birthday, I don't think this logic would capture them -- because it would be looking for their birthdate to be 18 years before midnight this morning.  I think the highAgeDate needs to be changed so the time component is 23:59:59.999999999, and the "lt" and "gt" comparators need to be "lte" and "gte".